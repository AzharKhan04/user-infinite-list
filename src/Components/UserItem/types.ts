export type Props = {
    name?: string,
    image_url?:string,
    email?:string,
    address?:string,
    backgroundColor?:string,
    profilePicName?:string,
    id?:any

};

type Types = {

    data:Props
}

export type IProps = Types;