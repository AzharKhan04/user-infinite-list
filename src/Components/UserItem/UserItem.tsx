import React from 'react';
import {IProps} from './types' 

const UserItem:React.FC<IProps>= ({data}) => {

    let {name,email,address,image_url,backgroundColor , profilePicName,id} = data


    return (
        <React.Fragment>
          <div  className="no-border card">
            <div className="card-body no-padding">
              <div style={{ alignItems: "center"}} className="row">
                <div className="col-2" style={{ marginTop: "12px" }}>
                  {image_url && (
                    <img
                      style={{ objectFit: "cover", borderRadius: "50%" }}
                      height="50"
                      width="50"
                      src={image_url}
                    />
                  )}
                  {!image_url && (
                    <div
                      style={{ backgroundColor: backgroundColor }}
                      className="defaultProfileSmall"
                    >
                      {profilePicName}
                    </div>
                  )}
                </div>
                <div className="col-7 text-truncate">
                  <div
                    style={{
                        
                      paddingTop: "20px",
                      color: "royalblue",
                    }}
                  >
                    {
                      <h5
                        style={{
                          fontSize: "16px",
                        }}
                      >
                        {name}
                      </h5>
                    }
                  </div>
                  <div
                    style={{
                      fontSize: "12px",
                      color: "grey",
                    }}
                  >
                    {<span>{email}</span>}
                  </div>
                  <div
                    style={{
                      fontSize: "12px",
                      color: "grey",
                    }}
                  >
                    {<span>{address}</span>}
                  </div>
                </div>
              </div>
            </div>
          </div>

        </React.Fragment>
      );



}
export default UserItem;