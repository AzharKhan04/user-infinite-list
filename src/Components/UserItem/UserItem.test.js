import renderer from 'react-test-renderer';
import UserItem from './UserItem';
import {Router} from 'react-router-dom'
import { render, screen } from '@testing-library/react';

describe('UserItem Component', () => {
    const testData = {
        name:'test user',
        email:'testemail@test.com',
        address:'testaddress',
        
    }

it('UserItem component renders correctly', () => {

  const location = { pathname:'' };
  
  const tree = renderer.create(<Router location={location}><UserItem data ={testData}/></Router>).toJSON();
  expect(tree).toMatchSnapshot();


});


test('UserItem renders data correctly', async () => {

    const location = { pathname:'' };
    render(<Router location={location}><UserItem data ={testData}/></Router>)
  
    expect(screen.getByText("test user")).toBeInTheDocument();
    expect(screen.getByText("testemail@test.com")).toBeInTheDocument();
    expect(screen.getByText("testaddress")).toBeInTheDocument();

     
  })
})