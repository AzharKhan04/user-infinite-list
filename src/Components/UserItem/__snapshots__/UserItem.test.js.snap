// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Component UserItem component renders correctly 1`] = `
<div
  className="no-border card"
>
  <div
    className="card-body no-padding"
  >
    <div
      className="row"
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <div
        className="col-2"
        style={
          Object {
            "marginTop": "12px",
          }
        }
      >
        <div
          className="defaultProfileSmall"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </div>
      <div
        className="col-7 text-truncate"
      >
        <div
          style={
            Object {
              "color": "royalblue",
              "paddingTop": "20px",
            }
          }
        >
          <h5
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            test user
          </h5>
        </div>
        <div
          style={
            Object {
              "color": "grey",
              "fontSize": "12px",
            }
          }
        >
          <span>
            testemail@test.com
          </span>
        </div>
        <div
          style={
            Object {
              "color": "grey",
              "fontSize": "12px",
            }
          }
        >
          <span>
            testaddress
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
